src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1413,9 +1413,9 @@ public SolutionState UpdateAdditionalDocumentTextLoader(DocumentId documentId, T
         {
             var oldDocument = GetRequiredAdditionalDocumentState(documentId);
 
-            // Assumes that text has changed. User could have closed a doc without saving and we are loading text from closed file with
-            // old content. Also this should make sure we don't re-use latest doc version with data associated with opened document.
-            return UpdateAdditionalDocumentState(oldDocument.UpdateText(loader, mode), textChanged: true, recalculateDependentVersions: true);
+            // Assumes that content has changed. User could have closed a doc without saving and we are loading text
+            // from closed file with old content.
+            return UpdateAdditionalDocumentState(oldDocument.UpdateText(loader, mode), contentChanged: true);
```

Comment:
What does contentChanged control here in the case of additional files? I wouldn't have imagined it mattered, but if it does, why isn't there one in the analyzer config method right below this one?

Reply:
so, in the end, it shouldn't matter as any attempt to get things like top-level versions will bail out immediately and return the normal text-version since these docs don't 'SupportSyntax'.  But i wanted to be consistent here around the concept that is happening.  There are doc changes that change contents, and doc changes that don't.  We should flow that along, and have any smarts happen later.

Reply:
Specifically, this will end up bottoming out here:

```c#
            if (_treeSource == null)
            {
                return await GetTextVersionAsync(cancellationToken).ConfigureAwait(false);
            }
```

So it's all ok.  But, for example, if we ever stored info with these docs, we'd want to know when they actually chnaged so we could properly dump it and recompute.